### PR TITLE
:lipstick: :computer: Increase Side Screen Size 

### LINKED ISSUE #450: :lipstick: :computer: Increase Side Screen Size
## User stories

As a user/org, I want to have more room to edit my blocks.

---
### Summary

Provide more room to edit my blocks. 

### Description:
At the moment the story editor page is split between the blocks library/edit forms portal and the main editor page. To achieve this all we have to do is make is make the side screen section toggleable. We'll do that by adding a button to our main editor page that will close and open the contents on side screen for us. When the edit btn on a block is clicked while its closed it will open the side screen section but show the edit forms instead of the blocks library.

The direction of the arrow in the button should switch depending on the direction of the side screen state

## Screenshot
SideScreen edit

### Acceptance Criteria

1. The Side screen section is toggleable. 

Please also note:
2. If a conversation about the issue I’m working on is happening,
the outcomes should be documented within that issue (not in the comments).

### PR DESCRIPTION
**Pull request by Michael-Otieno**
# Description

I worked alone.

This pull request solves issue #450, whereby I have redesigned the Block List UI following the design's guidelines.

Fixes #450

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Screenshot (optional)
https://user-images.githubusercontent.com/85155606/236444161-39ad5eff-0884-48ed-8b4f-0d7e1bcbc544.mp4

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
-  [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Please check the issue again, The requirement is for the side screen to toggleable when active

**Comment by Michael-Otieno** [comment]
Waiting for review @LemmyMwaura

**Comment by Michael-Otieno** [comment]
> Please check the issue again, The requirement is for the side screen to toggleable when active

Waiting for review @LemmyMwaura

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
There are a few situations you haven't taken into account.
- What happens when I click the edit btn on a block while the side screen is closed? (should open the side screen and also open the portal).
- What happens when I close the side screen while the form portal is open/attached. (should also be closed)

**Comment by Michael-Otieno** [comment]
> There are a few situations you haven't taken into account.
> 
> * What happens when I click the edit btn on a block while the side screen is closed? (should open the side screen and also open the portal).
> * What happens when I close the side screen while the form portal is open/attached. (should also be closed)

Updated PR. Waiting for review @LemmyMwaura

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
We have some performance concerns with the closing and opening. It lags while toggling this is because we're using ngIf. Challenge -> Try and see if you can optimize it.

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ import { _CreateKeywordJumpBlockMessageForm } from '../../model/keyword-jump-for
 import { _CreateEventBlockForm } from '../../model/event-block-form.model';
 
 import { BlockInjectorService } from '../../providers/block-injector.service';
+import { SideScreenToggleService } from 'libs/features/convs-mgr/stories/editor/src/lib/providers/side-screen-toggle.service';
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts:45]: use alised imports

Comment (Michael-Otieno) [reply libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts:45]: rectified

**Review by LemmyMwaura — APPROVED** [review]
Good Job 👍, I was hoping you'd use ngClass for the optimisation as it gives us more control over the design but I've switched it to that.

**Pull request merged**